crates/testsuite/tests/basic.rs

```diff
@@ -1,4 +1,4 @@
-use super::{dis_fn, val};
+use testsuite::{dis_fn, val};
```

Comment:
Still would prefer this to be in src, not tests, especially considering now that the crate name is literally called testsuite, but oh well.

(Context for others, having it be in tests instead of src builds 1 crate per file. I'm not sure if they're run simultaneously, but if they are, that's a problem, because the global lock is only per-process.)

Reply:
One of the advantages when structured like this though, is that all of the testing apparatus code is compiled once and shared with all tests, so adding or changing a test doesn't cause the entire testsuite to need to recompile, where it would if all of the tests were in `src`.

Reply:
Just want to check, I can't actually find any documentation on this, is it guaranteed each file/crate is run serially?

Reply:
I'm pretty sure in this case cargo is locking across processes, but I've reverted those changes for now and I'll write a bigger thing about testing in #48 as I think two are related.

Reply:
Cargo's lock does not help here, as tests can still modify files of the project outside of cargo's lock. In an earlier iteration of this framework, we were hitting failures due to that, which is why GLOBAL_MUTEX exists.